twtxt/helper.py

```diff
@@ -85,3 +87,28 @@ def sort_and_truncate_tweets(tweets, direction, limit):
             return sorted(tweets)
     else:
         return []
+
+
+@contextlib.contextmanager
+def atomic_write(path, mode):
+    """
+        Simple contextmanager to safely write to a
+        file.
+
+        :param str filename: the path to write to
+        :param str mode: the mode to open the file
+    """
+    # generate random filepath in same location as given filename
+    tmpfilepath = os.path.join(os.path.dirname(path), ".{}".format(uuid.uuid4()))
+    try:
+        if "a" in mode:
+            os.rename(path, tmpfilepath)
+
+        with open(tmpfilepath, mode) as atomicfile:
+            yield atomicfile
+        os.rename(tmpfilepath, path)
```

Comment:
I don't see your point. 
First I'm copying (if append mode) the existing tweets to a tempfile. After that I open it and yield the file object. The tweet can be written. After that the file will be closed and I'll copy the the new content back to the original location. Did I miss something?